depscan/lib/csaf.py

```diff
@@ -1141,52 +1021,72 @@
     r"bitbucket.org/[^\s/]+/[^\s/]+/wiki/": "Bitbucket Wiki Entry",
     r"https://vuldb.com/\?id.\d+": "VulDB Entry",
 }
-sorted_ref_map = dict(
-    sorted(ref_map.items(), key=lambda x: len(x[0]), reverse=True)
+SORTED_REF_MAP = dict(
+    sorted(REF_MAP.items(), key=lambda x: len(x[0]), reverse=True)
 )
 
-compiled_patterns = {
+COMPILED_REF_PATTERNS = {
     re.compile(pattern, re.IGNORECASE): value
-    for pattern, value in sorted_ref_map.items()
+    for pattern, value in SORTED_REF_MAP.items()
 }
 
+PURL_REGEX = re.compile(
+    r"(?P<pkg>pkg:([^/@?]+)/([^/@?]+)/?(?:[^/@?]+)?)@(?P<version>\w+(?:"
+    r"\.\w+\.)?(?:\w+\.?)?(?:\w+)?)(?P<type>\?type=\w+)?",
+    re.IGNORECASE,
+)
+
+ISSUES_REGEX = re.compile(
+    r"(?P<host>github|bitbucket|chromium)(?:.com|.org)/(?P<owner>["
+    r"\w\-.]+)/(?P<repo>[\w\-.]+)/issues/(?:detail\?id=)?(?P<id>\d+)",
+    re.IGNORECASE,
+)
+ADVISORY_REGEX = re.compile(
+    r"(?P<org>[^\s/.]+).(?:com|org)/(?:\S+/)*/?(?P<id>[\w\-:]+)",
+    re.IGNORECASE,
+)
+BUGZILLA_REGEX = re.compile(
+    r"(?<=bugzilla.)(?P<owner>\S+)\.\w{3}/show_bug.cgi\?id=(?P<id>\S+)",
+    re.IGNORECASE,
+)
+USN_REGEX = re.compile(
+    r"(?<=usn.ubuntu.com/)[\d\-]+|(?<=ubuntu.com/security/notices/USN-)"
+    r"[\d\-]+",
+    re.IGNORECASE,
+)
 
+
+@deprecated_class(comment="Use vdr_to_csaf function instead.", since="5.0.2")
```

Comment:
Can we remove instead of deprecate?

Reply:
@prabhu If you like... I am just a bit nervous about all the changes!